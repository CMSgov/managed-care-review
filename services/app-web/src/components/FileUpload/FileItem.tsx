import React from 'react'
import { Button } from '@trussworks/react-uswds'
import classnames from 'classnames'
import { SPACER_GIF } from './constants'

import styles from './FileUpload.module.scss'

export type FileStatus =
    | 'DUPLICATE_NAME_ERROR'
    | 'PENDING'
    | 'UPLOAD_COMPLETE'
    | 'UPLOAD_ERROR'
    | 'SAVED_TO_SUBMISSION'

// TODO: May want to use a union data type here to further clarify states
// FileItemUploadComplete | FileItemUploadIncomplete
export type FileItemT = {
    id: string
    name: string
    file?: File // only items that are not uploaded to s3 have this
    url?: string // only items uploaded to s3 have this
    key?: string // only items uploaded to s3 have this
    s3URL?: string // only items uploaded to s3 have this
    status: FileStatus
}

export type FileItemProps = {
    item: FileItemT
    deleteItem: (item: FileItemT) => void
    retryItem: (item: FileItemT) => void
}
export const FileItem = ({
    item,
    deleteItem,
    retryItem,
}: FileItemProps): React.ReactElement => {
    const { name, status } = item
    const hasDuplicateNameError = status === 'DUPLICATE_NAME_ERROR'
    const hasUploadError = status === 'UPLOAD_ERROR'
    const isLoading = status === 'PENDING'
    const isPDF = name.indexOf('.pdf') > 0
    const isWord = name.indexOf('.doc') > 0 || name.indexOf('.pages') > 0
    const isVideo = name.indexOf('.mov') > 0 || name.indexOf('.mp4') > 0
    const isExcel = name.indexOf('.xls') > 0 || name.indexOf('.numbers') > 0
    const isGeneric = !isPDF && !isWord && !isVideo && !isExcel

    const imageClasses = classnames('usa-file-input__preview-image', {
        'is-loading': isLoading,
        'usa-file-input__preview-image--pdf': isPDF,
        'usa-file-input__preview-image--word': isWord,
        'usa-file-input__preview-image--video': isVideo,
        'usa-file-input__preview-image--excel': isExcel,
        'usa-file-input__preview-image--generic': isGeneric,
    })

    const handleDelete = (e: React.MouseEvent) => {
        deleteItem(item)
    }

    const handleRetry = (e: React.MouseEvent) => {
        retryItem(item)
    }

    const Error = (): React.ReactElement | null => {
        if (hasDuplicateNameError)
            return (
                <>
                    <span className={styles.fileItemErrorMessage}>
                        Duplicate file
                    </span>
                    <span className={styles.fileItemErrorMessage}>
                        Please remove
                    </span>
                </>
            )
        else if (hasUploadError)
            return (
                <>
                    <span className={styles.fileItemErrorMessage}>
                        Upload failed
                    </span>
                    <span className={styles.fileItemErrorMessage}>
                        Please remove or retry
                    </span>
                </>
            )
        else {
            return null
        }
    }

    return (
        <>
            <div className={styles.fileItemText}>
                <img
                    id={item.id}
                    data-testid="file-input-preview-image"
                    src={SPACER_GIF}
                    alt=""
                    className={imageClasses}
                />
                <span
                    style={{
                        display: 'flex',
                        flexDirection: 'column',
                        fontSize: 'inherit',
                    }}
                >
                    <Error />
                    <span>{name}</span>
                </span>
            </div>
            <div className={styles.fileItemButtons}>
                <Button
                    type="button"
                    size="small"
                    unstyled
                    onClick={handleDelete}
                >
                    Remove
                </Button>
                {hasUploadError && (
                    <Button
                        type="button"
                        size="small"
                        unstyled
                        onClick={handleRetry}
                    >
                        Retry
                    </Button>
                )}
            </div>
        </>
    )
}
