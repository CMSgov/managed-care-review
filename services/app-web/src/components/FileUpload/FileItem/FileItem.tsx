import React from 'react'
import { Button } from '@trussworks/react-uswds'
import classnames from 'classnames'
import { SPACER_GIF } from '../constants'

import styles from '../FileUpload.module.scss'

export type FileStatus =
    | 'DUPLICATE_NAME_ERROR'
    | 'PENDING'
    | 'SCANNING'
    | 'SCANNING_ERROR'
    | 'UPLOAD_COMPLETE'
    | 'UPLOAD_ERROR'

export type FileItemT = {
    id: string
    name: string
    file?: File // only items that are not uploaded to s3 have this
    key?: string // only items uploaded to s3 have this
    s3URL?: string // only items uploaded to s3 have this
    status: FileStatus
}

const DocumentError = ({
    hasDuplicateNameError,
    hasScanningError,
    hasUploadError,
    hasUnexpectedError,
}: {
    hasDuplicateNameError: boolean
    hasScanningError: boolean
    hasUploadError: boolean
    hasUnexpectedError: boolean
}): React.ReactElement | null => {
    if (hasDuplicateNameError)
        return (
            <>
                <span className={styles.fileItemBoldMessage}>
                    Duplicate file
                </span>
                <span className={styles.fileItemBoldMessage}>
                    Please remove
                </span>
            </>
        )
    else if (hasScanningError && !hasUnexpectedError)
        return (
            <>
                <span className={styles.fileItemBoldMessage}>
                    Failed security scan, please remove
                </span>
            </>
        )
    else if (hasUploadError && !hasUnexpectedError)
        return (
            <>
                <span className={styles.fileItemBoldMessage}>
                    Upload failed
                </span>
                <span className={styles.fileItemBoldMessage}>
                    Please remove or retry
                </span>
            </>
        )
    else if (hasUnexpectedError) {
        return (
            <>
                <span className={styles.fileItemBoldMessage}>
                    Upload failed
                </span>
                <span className={styles.fileItemBoldMessage}>
                    Unexpected error. Please remove.
                </span>
            </>
        )
    } else {
        return null
    }
}

type FileItemProps = {
    item: FileItemT
    deleteItem: (item: FileItemT) => void
    retryItem: (item: FileItemT) => void
}
export const FileItem = ({
    item,
    deleteItem,
    retryItem,
}: FileItemProps): React.ReactElement => {
    const { name, status, file } = item
    const hasDuplicateNameError = status === 'DUPLICATE_NAME_ERROR'
    const hasScanningError = status === 'SCANNING_ERROR'
    const hasUploadError = status === 'UPLOAD_ERROR'
    const hasUnexpectedError = status === 'UPLOAD_ERROR' && file === undefined
    const isLoading = status === 'PENDING'
    const isScanning = status === 'SCANNING'

    const isPDF = name.indexOf('.pdf') > 0
    const isWord = name.indexOf('.doc') > 0 || name.indexOf('.pages') > 0
    const isVideo = name.indexOf('.mov') > 0 || name.indexOf('.mp4') > 0
    const isExcel = name.indexOf('.xls') > 0 || name.indexOf('.numbers') > 0
    const isGeneric = !isPDF && !isWord && !isVideo && !isExcel

    const imageClasses = classnames('usa-file-input__preview-image', {
        'is-loading': isLoading || isScanning,
        'usa-file-input__preview-image--pdf': isPDF,
        'usa-file-input__preview-image--word': isWord,
        'usa-file-input__preview-image--video': isVideo,
        'usa-file-input__preview-image--excel': isExcel,
        'usa-file-input__preview-image--generic': isGeneric,
    })

    const handleDelete = (_e: React.MouseEvent) => {
        deleteItem(item)
    }

    const handleRetry = (_e: React.MouseEvent) => {
        retryItem(item)
    }

    return (
        <>
            <div className={styles.fileItemText}>
                <img
                    id={item.id}
                    data-testid="file-input-preview-image"
                    src={SPACER_GIF}
                    alt=""
                    className={imageClasses}
                />
                <span
                    style={{
                        display: 'flex',
                        flexDirection: 'column',
                        fontSize: 'inherit',
                    }}
                >
                    <DocumentError
                        hasDuplicateNameError={hasDuplicateNameError}
                        hasScanningError={hasScanningError}
                        hasUploadError={hasUploadError}
                        hasUnexpectedError={hasUnexpectedError}
                    />
                    <>
                        {(isLoading || isScanning) && (
                            <span className={styles.fileItemBoldMessage}>
                                {isLoading
                                    ? 'Step 1 of 2: Uploading'
                                    : 'Step 2 of 2: Scanning'}
                            </span>
                        )}
                        <span>{name}</span>
                    </>
                </span>
            </div>
            <div className={styles.fileItemButtons}>
                <Button
                    type="button"
                    size="small"
                    unstyled
                    onClick={handleDelete}
                >
                    Remove
                </Button>
                {(hasUploadError || hasScanningError) && !hasUnexpectedError && (
                    <Button
                        type="button"
                        size="small"
                        unstyled
                        onClick={handleRetry}
                    >
                        Retry
                    </Button>
                )}
            </div>
        </>
    )
}
