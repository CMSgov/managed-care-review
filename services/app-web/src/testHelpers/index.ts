export {
    getExternalInputFromDateRange,
    getYesNoFieldValue,
    getParent,
} from './fieldHelpers'
export {
    fakeRequest,
    dragAndDrop,
    renderWithProviders,
    prettyDebug,
    userClickByRole,
    userClickByTestId,
    userClickSignIn,
    TEST_DOC_FILE,
    TEST_DOCX_FILE,
    TEST_PDF_FILE,
    TEST_PNG_FILE,
    TEST_TEXT_FILE,
    TEST_VIDEO_FILE,
    TEST_XLS_FILE,
} from './jestHelpers'

export { testS3Client } from './s3Helpers'
export { updateDateRange } from './userEventHelpers'
