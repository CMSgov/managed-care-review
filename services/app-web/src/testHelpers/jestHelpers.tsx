import { MockedProvider, MockedProviderProps } from '@apollo/client/testing'
import { MemoryRouter } from 'react-router-dom'
import {
    fireEvent,
    render,
    Screen,
    queries,
    ByRoleMatcher,
} from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { AuthProvider, AuthProviderProps } from '../contexts/AuthContext'

import { PageProvider } from '../contexts/PageContext'
import { S3Provider } from '../contexts/S3Context'
import { testS3Client } from './s3Helpers'

/* Render */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const renderWithProviders = (
    ui: React.ReactNode,
    options?: {
        routerProvider?: { route?: string }
        apolloProvider?: MockedProviderProps
        authProvider?: Partial<AuthProviderProps>
    }
) => {
    const {
        routerProvider = {},
        apolloProvider = {},
        authProvider = {},
    } = options || {}

    const { route } = routerProvider

    return render(
        <MockedProvider {...apolloProvider}>
            <MemoryRouter initialEntries={[route || '']}>
                <AuthProvider authMode={'AWS_COGNITO'} {...authProvider}>
                    <S3Provider client={testS3Client}>
                        <PageProvider>{ui}</PageProvider>
                    </S3Provider>
                </AuthProvider>
            </MemoryRouter>
        </MockedProvider>
    )
}

/* User Events */

const userClickByTestId = (
    screen: Screen<typeof queries>,
    testId: string
): void => {
    const element = screen.getByTestId(testId)
    userEvent.click(element)
}
const userClickByRole = (
    screen: Screen<typeof queries>,
    role: ByRoleMatcher,
    options?: queries.ByRoleOptions | undefined
): void => {
    const element = screen.getByRole(role, options)
    userEvent.click(element)
}

const userClickSignIn = (screen: Screen<typeof queries>): void => {
    const signInButton = screen.getByRole('link', { name: /Sign In/i })
    userEvent.click(signInButton)
}

function dragAndDrop(inputDropTarget: HTMLElement, files: File[]): void {
    fireEvent.dragEnter(inputDropTarget)
    fireEvent.dragOver(inputDropTarget)
    fireEvent.drop(inputDropTarget, {
        bubbles: true,
        cancelable: true,
        dataTransfer: {
            files: files,
        },
    })

    return
}

function fakeRequest<T>(
    success: boolean,
    returnData: T,
    timeout?: number
): Promise<T> {
    const t = timeout || Math.round(Math.random() * 1000)
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            if (success) {
                resolve(returnData)
            } else {
                reject(new Error('Error'))
            }
        }, t)
    })
}

const TEST_TEXT_FILE = new File(['Test File Contents'], 'testFile.txt', {
    type: 'text/plain',
})

const TEST_PDF_FILE = new File(['Test PDF File'], 'testFile.pdf', {
    type: 'application/pdf',
})

const TEST_DOC_FILE = new File(['Test doc File'], 'testFile.doc', {
    type: 'application/msword',
})

const TEST_XLS_FILE = new File(['Test xls File'], 'testFile.xls', {
    type: 'application/vnd.ms-excel',
})

const TEST_VIDEO_FILE = new File(['Test video File'], 'testFile.mp4', {
    type: 'video/mp4',
})

const TEST_PNG_FILE = new File(['Test PNG Image'], 'testFile.png', {
    type: 'image/png',
})

export {
    fakeRequest,
    dragAndDrop,
    renderWithProviders,
    userClickByRole,
    userClickByTestId,
    userClickSignIn,
    TEST_DOC_FILE,
    TEST_PDF_FILE,
    TEST_PNG_FILE,
    TEST_TEXT_FILE,
    TEST_VIDEO_FILE,
    TEST_XLS_FILE,
}
