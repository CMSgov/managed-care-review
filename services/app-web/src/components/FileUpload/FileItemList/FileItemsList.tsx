import React from 'react'
import classnames from 'classnames'
import { Table } from '@trussworks/react-uswds'
import { FileItem, FileItemT, FileStatus } from '../FileItem/FileItem'
import styles from '../FileUpload.module.scss'

export const FileItemsList = ({
    fileItems,
    deleteItem,
    retryItem,
    renderMode,
}: {
    fileItems: FileItemT[]
    deleteItem: (id: FileItemT) => void
    retryItem: (item: FileItemT) => void
    renderMode: 'table' | 'list'
}): React.ReactElement => {
    const liClasses = (status: FileStatus): string => {
        const hasError =
            status === 'UPLOAD_ERROR' ||
            status === 'SCANNING_ERROR' ||
            status === 'DUPLICATE_NAME_ERROR'
        return classnames(styles.fileItem, {
            'bg-secondary-lighter border-secondary margin-top-1px': hasError,
            'usa-file-input__preview': !hasError,
        })
    }

    if (renderMode === 'table') {
        return (
            <Table fullWidth>
                <thead>
                    <tr>
                        <th>Document name</th>
                        <th>Date uploaded</th>
                        <th></th>
                    </tr>
                </thead>
                <tbody>
                    {fileItems.map((item) => (
                        <FileItem
                            key={item.id}
                            deleteItem={deleteItem}
                            retryItem={retryItem}
                            item={item}
                            renderMode="table"
                        />
                    ))}
                </tbody>
            </Table>
        )
    } else {
        return (
            <ul
                data-testid="file-input-preview-list"
                className={styles.fileItemsList}
            >
                {fileItems.map((item) => (
                    <li
                        key={item.id}
                        id={item.id}
                        className={liClasses(item.status)}
                    >
                        <FileItem
                            deleteItem={deleteItem}
                            retryItem={retryItem}
                            item={item}
                            renderMode="list"
                        />
                    </li>
                ))}
            </ul>
        )
    }
}
