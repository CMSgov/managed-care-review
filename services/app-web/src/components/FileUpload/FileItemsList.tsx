import React from 'react'
import classnames from 'classnames'
import styles from './FileUpload.module.scss'
import { FileItem, FileItemT, FileStatus } from './FileItem'

export const FileItemsList = ({
    fileItems,
    deleteItem,
    retryItem,
}: {
    fileItems: FileItemT[]
    deleteItem: (id: FileItemT) => void
    retryItem: (item: FileItemT) => void
}): React.ReactElement => {
    const liClasses = (status: FileStatus): string => {
        const hasError =
            status === 'UPLOAD_ERROR' || status === 'DUPLICATE_NAME_ERROR'
        return classnames(styles.fileItem, {
            'bg-secondary-lighter border-secondary margin-top-1px': hasError,
            'usa-file-input__preview': !hasError,
        })
    }

    return (
        <ul
            data-testid="file-input-preview-list"
            className={styles.fileItemList}
        >
            {fileItems.map((item) => (
                <li
                    key={item.id}
                    id={item.id}
                    className={liClasses(item.status)}
                >
                    <FileItem
                        deleteItem={deleteItem}
                        retryItem={retryItem}
                        item={item}
                    />
                </li>
            ))}
        </ul>
    )
}
