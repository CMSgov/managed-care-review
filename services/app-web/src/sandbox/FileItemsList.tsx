import React from 'react'
import { FileItem, FileItemT } from './FileItem'

export const FileItemsList = ({
    fileItems,
    deleteItem,
}: {
    fileItems: FileItemT[]
    deleteItem: (id: string) => void
}): React.ReactElement => {
    return (
        <ul
            style={{
                listStyleType: 'none',
                display: 'inline-block',
                padding: 0,
                margin: '0 0 -1px ',
                width: '480px',
            }}
        >
            {fileItems.map((item) => (
                <li
                    key={item.id}
                    id={item.id}
                    className={
                        item.status === 'UPLOAD_ERROR'
                            ? 'bg-secondary-lighter border-secondary'
                            : 'usa-file-input__preview'
                    }
                    style={{
                        display: 'flex',
                        justifyContent: 'space-between',
                        pointerEvents: 'all',
                    }}
                >
                    <FileItem deleteItem={deleteItem} item={item} />
                </li>
            ))}
        </ul>
    )
}
