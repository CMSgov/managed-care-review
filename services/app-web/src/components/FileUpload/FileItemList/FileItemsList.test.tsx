import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import userEvent from '@testing-library/user-event'

import { FileItemT } from '../FileItem/FileItem'
import { FileItemsList } from './FileItemsList'
import { TEST_PDF_FILE } from '../../../testHelpers/jestHelpers'

describe('FileItemList component', () => {
    const pending: FileItemT = {
        id: 'testFile',
        file: TEST_PDF_FILE,
        name: 'testFile.pdf',
        key: undefined,
        s3URL: undefined,
        status: 'PENDING',
    }
    const scanning: FileItemT = {
        id: 'testFile1',
        file: TEST_PDF_FILE,
        name: 'testFile1.pdf',
        key: '4545454-testFile1',
        s3URL: 'tests3://uploaded-12313123213/4545454-testFile1',
        status: 'SCANNING',
    }
    const uploadError: FileItemT = {
        id: 'testFile2',
        file: TEST_PDF_FILE,
        name: 'testFile2.pdf',
        key: undefined,
        s3URL: undefined,
        status: 'UPLOAD_ERROR',
    }

    const scanningError: FileItemT = {
        id: 'testFile3',
        file: TEST_PDF_FILE,
        name: 'testFile3.pdf',
        key: '4545454-testFile3',
        s3URL: 'tests3://uploaded-12313123213/4545454-testFile3',
        status: 'SCANNING_ERROR',
    }

    const complete: FileItemT = {
        id: 'testFile4',
        file: TEST_PDF_FILE,
        name: 'testFile4.pdf',
        key: '4545454-testFile4',
        s3URL: 'tests3://uploaded-12313123213/4545454-testFile4',
        status: 'UPLOAD_COMPLETE',
    }

    const duplicateError: FileItemT = {
        id: 'testFile44',
        file: TEST_PDF_FILE,
        name: 'testFile4.pdf',
        key: '1234545454-testFile4',
        s3URL: 'tests3://uploaded-12313123213/1234545454-testFile4',
        status: 'DUPLICATE_NAME_ERROR',
    }
    const buttonActionProps = {
        deleteItem: jest.fn(),
        retryItem: jest.fn(),
    }

    beforeEach(() => jest.clearAllMocks())
    it('renders without errors', () => {
        const fileItems = [pending, uploadError]
        render(
            <FileItemsList
                renderMode="list"
                fileItems={fileItems}
                {...buttonActionProps}
            />
        )

        expect(screen.getAllByRole('listitem').length).toEqual(fileItems.length)
        expect(screen.getByText(/testFile.pdf/)).toBeInTheDocument()
        expect(screen.getByText('testFile2.pdf')).toBeInTheDocument()
    })

    it('renders correctly', () => {
        const fileItems = [
            pending,
            uploadError,
            scanningError,
            complete,
            duplicateError,
        ]
        const tree = renderer
            .create(
                <FileItemsList
                    renderMode="list"
                    fileItems={fileItems}
                    {...buttonActionProps}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('button actions work as expected', () => {
        const fileItems = [uploadError]
        render(
            <FileItemsList
                renderMode="list"
                fileItems={fileItems}
                {...buttonActionProps}
            />
        )

        userEvent.click(screen.getByRole('button', { name: 'Retry' }))
        expect(buttonActionProps.retryItem).toHaveBeenCalled()

        userEvent.click(screen.getByRole('button', { name: 'Remove' }))
        expect(buttonActionProps.deleteItem).toHaveBeenCalled()
    })

    it('displays error styles for items that have errors', () => {
        const fileItems = [
            pending,
            uploadError,
            scanningError,
            complete,
            duplicateError,
            scanning,
        ]
        render(
            <FileItemsList
                renderMode="list"
                fileItems={fileItems}
                {...buttonActionProps}
            />
        )

        const listItems = screen.getAllByRole('listitem')
        const loadingListItem = listItems[0]
        const uploadErrorListItem = listItems[1]
        const scanningErrorListItem = listItems[2]
        const completeListItem = listItems[3]
        const duplicateErrorListItem = listItems[4]
        const scanningListItem = listItems[5]

        // Items not in error state
        expect(loadingListItem).toHaveClass('usa-file-input__preview')
        expect(scanningListItem).toHaveClass('usa-file-input__preview')
        expect(completeListItem).toHaveClass('usa-file-input__preview')

        // Items in an error state
        expect(uploadErrorListItem).not.toHaveClass('usa-file-input__preview')
        expect(uploadErrorListItem).toHaveClass(
            'bg-secondary-lighter border-secondary '
        )
        expect(scanningErrorListItem).not.toHaveClass('usa-file-input__preview')
        expect(scanningErrorListItem).toHaveClass(
            'bg-secondary-lighter border-secondary '
        )
        expect(completeListItem).toHaveClass('usa-file-input__preview')
        expect(duplicateErrorListItem).not.toHaveClass(
            'usa-file-input__preview'
        )
        expect(duplicateErrorListItem).toHaveClass(
            'bg-secondary-lighter border-secondary '
        )
    })
})
