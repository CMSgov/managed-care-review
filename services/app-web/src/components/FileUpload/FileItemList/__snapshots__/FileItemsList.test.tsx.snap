// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileItemList component renders correctly 1`] = `
<ul
  className="fileItemsList"
  data-testid="file-input-preview-list"
  id="file-items-list"
  tabIndex={-1}
>
  <li
    className="fileItem usa-file-input__preview"
    id="testFile"
  >
    <div
      className="fileItemText"
    >
      <div
        aria-label="Status of file testFile.pdf"
        aria-valuetext="uploading"
        role="progressbar"
      >
        <img
          alt=""
          className="usa-file-input__preview-image is-loading usa-file-input__preview-image--pdf"
          data-testid="file-input-preview-image"
          id="testFile"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
      </div>
      <span
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "fontSize": "inherit",
          }
        }
      >
        <span
          className="fileItemBoldMessage"
        >
          Step 1 of 2: Uploading
        </span>
        <span>
          testFile.pdf
        </span>
      </span>
    </div>
    <div
      className="fileItemButtons"
    >
      <button
        aria-label="Remove testFile.pdf document"
        className="usa-button usa-button--small usa-button--unstyled"
        data-testid="button"
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </div>
  </li>
  <li
    className="fileItem bg-secondary-lighter border-secondary margin-top-1px"
    id="testFile2"
  >
    <div
      className="fileItemText"
    >
      <div
        aria-label="Status of file testFile2.pdf"
        aria-valuetext="error"
        role="progressbar"
      >
        <img
          alt=""
          className="usa-file-input__preview-image usa-file-input__preview-image--pdf"
          data-testid="file-input-preview-image"
          id="testFile2"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
      </div>
      <span
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "fontSize": "inherit",
          }
        }
      >
        <span
          className="fileItemBoldMessage"
        >
          Upload failed
        </span>
        <span
          className="fileItemBoldMessage"
        >
          Please remove or retry
        </span>
        <span>
          testFile2.pdf
        </span>
      </span>
    </div>
    <div
      className="fileItemButtons"
    >
      <button
        aria-label="Remove testFile2.pdf document"
        className="usa-button usa-button--small usa-button--unstyled"
        data-testid="button"
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
      <button
        aria-label="Retry upload for testFile2.pdf document"
        className="usa-button usa-button--small usa-button--unstyled"
        data-testid="button"
        onClick={[Function]}
        type="button"
      >
        Retry
      </button>
    </div>
  </li>
  <li
    className="fileItem bg-secondary-lighter border-secondary margin-top-1px"
    id="testFile3"
  >
    <div
      className="fileItemText"
    >
      <div
        aria-label="Status of file testFile3.pdf"
        aria-valuetext="error"
        role="progressbar"
      >
        <img
          alt=""
          className="usa-file-input__preview-image usa-file-input__preview-image--pdf"
          data-testid="file-input-preview-image"
          id="testFile3"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
      </div>
      <span
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "fontSize": "inherit",
          }
        }
      >
        <span
          className="fileItemBoldMessage"
        >
          Failed security scan, please remove
        </span>
        <span>
          testFile3.pdf
        </span>
      </span>
    </div>
    <div
      className="fileItemButtons"
    >
      <button
        aria-label="Remove testFile3.pdf document"
        className="usa-button usa-button--small usa-button--unstyled"
        data-testid="button"
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
      <button
        aria-label="Retry upload for testFile3.pdf document"
        className="usa-button usa-button--small usa-button--unstyled"
        data-testid="button"
        onClick={[Function]}
        type="button"
      >
        Retry
      </button>
    </div>
  </li>
  <li
    className="fileItem usa-file-input__preview"
    id="testFile4"
  >
    <div
      className="fileItemText"
    >
      <div
        aria-label="Status of file testFile4.pdf"
        aria-valuetext=""
        role="progressbar"
      >
        <img
          alt=""
          className="usa-file-input__preview-image usa-file-input__preview-image--pdf"
          data-testid="file-input-preview-image"
          id="testFile4"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
      </div>
      <span
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "fontSize": "inherit",
          }
        }
      >
        <span>
          testFile4.pdf
        </span>
      </span>
    </div>
    <div
      className="fileItemButtons"
    >
      <button
        aria-label="Remove testFile4.pdf document"
        className="usa-button usa-button--small usa-button--unstyled"
        data-testid="button"
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </div>
  </li>
  <li
    className="fileItem bg-secondary-lighter border-secondary margin-top-1px"
    id="testFile44"
  >
    <div
      className="fileItemText"
    >
      <div
        aria-label="Status of file testFile4.pdf"
        aria-valuetext="error"
        role="progressbar"
      >
        <img
          alt=""
          className="usa-file-input__preview-image usa-file-input__preview-image--pdf"
          data-testid="file-input-preview-image"
          id="testFile44"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
      </div>
      <span
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "fontSize": "inherit",
          }
        }
      >
        <span
          className="fileItemBoldMessage"
        >
          Duplicate file
        </span>
        <span
          className="fileItemBoldMessage"
        >
          Please remove
        </span>
        <span>
          testFile4.pdf
        </span>
      </span>
    </div>
    <div
      className="fileItemButtons"
    >
      <button
        aria-label="Remove testFile4.pdf document"
        className="usa-button usa-button--small usa-button--unstyled"
        data-testid="button"
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </div>
  </li>
</ul>
`;
